client/components/explorer/sidebar/sidebar-tab-service_test.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * @copyright Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @fileoverview Tests for SidebarTabService, holds state and data for the
+ * explorer sidebar.
+ * 
+ * @author joemu@google.com (Joe Allan Muharsky)
+ */
+
+goog.require('p3rf.perfkit.explorer.components.config.ConfigService');
+goog.require('p3rf.perfkit.explorer.components.dashboard.DashboardService');
+goog.require('p3rf.perfkit.explorer.components.explorer.sidebar.SidebarTabService');
+goog.require('p3rf.perfkit.explorer.components.explorer.sidebar.SidebarTabModel');
+
+describe('SidebarTabService', function() {
+  var scope, sidebarTabSvc;
+
+  const explorer = p3rf.perfkit.explorer;
+  const SidebarTabModel = explorer.components.explorer.sidebar.SidebarTabModel;
+
+  var mockTabs = [
+    {id: 'global-a', title: 'Global Tab 1', iconClass: 'global-a-icon',
+     hint: 'Tip for Global Tab 1', requireWidget: false,
+     tabClass: 'global-a-tab', panelTitleClass: 'global-a-panel-title',
+     panelClass: 'global-a-panel'},
+    {id: 'widget-b', title: 'Widget Tab 2', iconClass: 'widget-b-icon',
+     hint: 'Hint for Widget Tab 2', requireWidget: true,
+     tabClass: 'widget-b-tab', panelTitleClass: 'widget-b-panel-title',
+     panelClass: 'widget-b-panel'},
+    {id: 'global-b', title: 'Global Tab 2', iconClass: 'global-b-icon',
+     hint: 'Hint for Global Tab 2', requireWidget: false,
+     tabClass: 'global-b-tab', panelTitleClass: 'global-b-panel-title',
+     panelClass: 'global-b-panel'},
+    {id: 'widget-a', title: 'Widget Tab a', iconClass: 'widget-a-icon',
+     hint: 'Hint for Widget Tab a', requireWidget: true,
+     tabClass: 'widget-a-tab', panelTitleClass: 'widget-a-panel-title',
+     panelClass: 'widget-a-panel'}
+  ];
+
+  beforeEach(module('explorer'));
+
+  beforeEach(inject(function(_sidebarTabService_) {
+    sidebarTabSvc = _sidebarTabService_;
+  }));
+
+  describe('should correctly initialize', function() {
+    it('the tabs list with default values', function() {
+      expect(sidebarTabSvc.tabs).not.toBeNull();
+      expect(sidebarTabSvc.tabs).toBeArrayOfSize(6);
```

Comment:
To make this a little less brittle when tabs are added / removed, consider making the tab list constant in sidebar-tab-service.js, and checking the size against that.